--- ./default/js/dojo18/mpi/vp/SearchView.js	2019-04-18 16:02:36.000000000 -0400
+++ ./custom/js/dojo18/mpi/vp/SearchView.js	2019-04-18 16:02:34.000000000 -0400
@@ -24,10 +24,38 @@
 ], function(View, _Templated, i18n, dom, domStyle, domClass, domConstruct, query, lang, declare, has, sniff, TooltipDialog, popup, restXhr, eventHandling, template, keys, string, on, queryj) {
 	
 	var RC = i18n.getLocalization("mpi.vp", "common", i18n.normalizeLocale());
+
+	var path = "";
+	var search = document.location.search;
+	var pathParts = document.location.pathname.split("/");
+	if(pathParts.length > 0) {
+		path = pathParts.pop();
+	}
+
+	function updateSubnavigation(){
+		var hamburger = document.getElementById("gskHamburger")
+		if(!document.body || !hamburger) {
+			return setTimeout(updateSubnavigation, 250);
+		}
+		if(path === "search.vp" || path === "video.vp") {
+			domClass.add(document.body, "hide-subnavigation");
+		} else {
+			domClass.remove(document.body, "hide-subnavigation");
+		}
+		if((path === "home.vp" || path === "channel.vp") && !has("qumu-phone") && !has("qumu-tablet")) {
+			domClass.add(hamburger, "active");
+			domClass.add(document.body, "show-sidebar");		 
+		}
+		var footer = document.createElement("div");
+		footer.id = "gskFooter";
+		footer.innerHTML = '<div class="footerLabel">GSK Tube</div><div class="footerList"><a href="https://cbshelp.gsk.com/Pages/GSKTube-Terms-and-Conditions.aspx">GSK Tube terms and conditions</a><a href="https://myconnect.gsk.com/sites/cGSKAssets/pages/UseAndAccessPolicies.aspx">Your responsibilities online</a><a href="https://connect.gsk.com/sites/scs/CITC/CCC/GSKTube/Documents/Branded%20Consent%20to%20record.docx">Consent request form</a><a href="https://fb.me/g/9Sb3Voelh/uyt3wZrx">Contacts</a><a href="https://cbshelp.gsk.com/Pages/GSK-Tube-Record-Retention-Policy.aspx">GSK Tube record retention policy</a></div>';
+    document.body.appendChild(footer);
+	}
+	updateSubnavigation();
+
 	
 	return declare( "mpi.vp.SearchView", [View, _Templated],
 	{
-
 		RC: RC,
 		allowLogin: true,
 		templateString: template,
@@ -45,6 +73,11 @@
 		defaultSearch: null,
 		isAutoSuggestionEnabled: false,
 		params : null, 
+		homeActive: path === "home.vp" ? "active" : '',
+		favoritesActive: path === "myfavorites.vp" ? "active" : '',
+		broadcastsActive: path === "mywebcasts.vp" ? "active" : '',
+		uploadActive: path === "myprograms.vp" ? "active" : '',
+		supportActive: path + search === "channel.vp?channelId=esc_prg_cntr_chn:2502" ? "active" : '',
 		postCreate: function() {
 			this.inherited(arguments);
 			
@@ -64,7 +97,7 @@
 			this.toolTipContainer.appendChild(this.dialog.domNode);
 			
 			if(this.dialog.containerNode.parentElement.parentElement) {
-				domClass.add(this.dialog.containerNode.parentElement.parentElement,"accessPaneDropDown");				
+				domClass.add(this.dialog.containerNode.parentElement.parentElement,"accessPaneDropDown");				 
 			}
 			
 			on(window, "resize", lang.hitch(this, this._onWindowResize));
@@ -170,7 +203,7 @@
 		_onWindowResize: function(e) {
 			if(domClass.contains(this.accessButton,"active")) {
 				this._openAccessPanel(e);
-				this._openAccessPanel(e);				
+				this._openAccessPanel(e);				 
 			}
 			if(this.isAutoSuggestionEnabled && !domClass.contains(this.possibleSearchTermsNode, "hide")) {
 				domClass.add(this.possibleSearchTermsNode, "hide");
@@ -231,8 +264,7 @@
 			if(this.defaultSearch == true) e.target.value = "";
 			this.searchInput.focus();
 			
-			if(has("ie") && domClass.contains(document.body,"tabFocus")) this.searchInput.select();
-			
+			if(has("ie") && domClass.contains(document.body,"tabFocus")) this.searchInput.select();			 
 			this.initiateAutoSuggest();
 		},
 		
@@ -261,7 +293,7 @@
 				this.defaultSearch = true;
 			}
 		},
-		
+	
 		initiateAutoSuggest: function() {
 			if(!has("qumu-phone") && this.isAutoSuggestionEnabled) {
 				//if(!this.sendToServer || (this.sendToServer && this.sendToServer == this.searchInput.value && !domClass.contains(this.possibleSearchTermsNode, "hide"))) {
@@ -329,7 +361,7 @@
 				for(var i=0; i<this.possibleSearchTerms.length; i++) {
 					searchTerm = domConstruct.create("div",{"class": "possibleSearchTerm", "data": this.possibleSearchTerms[i].prefix + this.possibleSearchTerms[i].suggestion}, d,"last");
 					if(this.possibleSearchTerms[i].prefix.length > 0) {
-						domConstruct.create("span",{"class": "prefix", "innerHTML": this.possibleSearchTerms[i].prefix, "data": this.possibleSearchTerms[i].prefix + this.possibleSearchTerms[i].suggestion}, searchTerm,"last");					
+						domConstruct.create("span",{"class": "prefix", "innerHTML": this.possibleSearchTerms[i].prefix, "data": this.possibleSearchTerms[i].prefix + this.possibleSearchTerms[i].suggestion}, searchTerm,"last");					 
 					}
 					//this.possibleSearchTerms[i].suggestion = this.possibleSearchTerms[i].suggestion + "garbage_garbage_garbage_garbage_garbage_garbage_garbage_garbage_garbage_garbage_";
 					domConstruct.create("span",{"class": "autoCompleteTerm", "innerHTML": this.possibleSearchTerms[i].suggestion, "data": this.possibleSearchTerms[i].prefix + this.possibleSearchTerms[i].suggestion}, searchTerm,"last");
@@ -374,7 +406,7 @@
 					}
 				});
 				
-				searchTermWidth = getEleWidth(d.getBoundingClientRect());					
+				searchTermWidth = getEleWidth(d.getBoundingClientRect());					 
 				for(var i = 0; i < d.children.length; i++) {
 					childrenWidth += getEleWidth(d.children[i].getBoundingClientRect());
 				}
@@ -479,7 +511,7 @@
 					} else {
 						this.autoSuggestPointer = -1;
 						this.searchInput.value = this.sendToServer;
-					}					
+					}					 
 				}));
 				return true;
 			}
@@ -503,7 +535,7 @@
 		},
 		_search: function() {
 			if(!has("qumu-phone")) {
-				this._searchIt();	
+				this._searchIt();	 
 			} else {
 				this._showOrHidePhoneSearchBox();
 			}
@@ -532,6 +564,7 @@
 			}
 			this.params= filteredParams;
 		},
+
 		_searchIt: function() {
 			
 			var query = this.searchInput.value;
@@ -543,14 +576,14 @@
 			}
 			if(this.threshold>0)	query = query + '&threshold=' + this.threshold;
 			if(this.drill)	query = query + '&drill=' + encodeURIComponent(this.drill);
+
 			if(this.params.length > 0)
 				{
 					this._removeOtherParams();
 					document.location.href='search.vp?query='+ query+this.params;
-				}else
-					{
+				} else {
 					document.location.href='search.vp?query='+ query;
-					}
+				}
 		},
 		
 		_showOrHidePhoneSearchBox: function() {
@@ -594,6 +627,19 @@
 			}
 		},
 		
+		_onHamburgerClick: function(e) {
+			if(e && e.type=="keydown") {
+				if(e.keyCode != 13)
+					return;
+				else {
+					e.preventDefault();
+					e.stopPropagation();
+				}
+			}
+			domClass.toggle(this.hamburger, "active");
+			domClass.toggle(document.body, "show-sidebar");
+		},
+
 		_onGoClick: function(e) {
 			if(e && e.type=="keydown") {
 				if(e.keyCode != 13)
