--- ./default/js/dojo18/mpi/util/DateTimeWidget.js	2019-04-18 16:02:36.000000000 -0400
+++ ./custom/js/dojo18/mpi/util/DateTimeWidget.js	2019-04-18 16:02:34.000000000 -0400
@@ -61,7 +61,7 @@
 				name: _this.name + "_time",
 				value: val,
 				style: _this.timeStyle,
-				'aria-label': this.context
+				'aria-label': this.contexts
 			}, _this._timePicker);
 			domClass.add(_this._timePicker.domNode, "timePick");
 			_this._timePicker.watch("value", function(prop, oldValue, newValue) {
@@ -104,9 +104,9 @@
 				return;
 			}
 
-			this.toolTip = new toolTip({   
+			this.toolTip = new toolTip({	 
 				label: val,
-				showDelay: 250,     
+				showDelay: 250,			
 				connectId: this.dtwInner,
 				position: ['after','below']
 			}); 
